flink-runtime/src/main/java/org/apache/flink/runtime/state/filesystem/FsCheckpointStreamFactory.java

```diff
@@ -266,6 +264,16 @@ public void flush() throws IOException {
 			}
 		}
 
+		/**
+		 * Flush buffers to file if their size is above {@link #localStateThreshold}.
+		 */
+		@Override
+		public void flush() throws IOException {
+			if (pos > localStateThreshold) {
```

Comment:
Maybe extend this to `if (stream != null || pos > localStateThreshold)`?

Reply:
Makes sense, expanded the condition.

Reply:
I guess the added condition of `stream != null` is not addressed yet?

Reply:
I must have reverted it while updating the tests. Fixed.